contracts/tokens.js

```diff
@@ -105,6 +108,198 @@ actions.createSSC = async () => {
       }
     }
   }
+
+  /**
+   * Adds the table to store authorized contracts to do certain actions, i.e. transfer
+   */
+  tableExists = await api.db.tableExists('authorizations');
+  if (tableExists === false) {
+    await api.db.createTable('authorizations', ['account', 'contract', 'version', 'symbol', 'action', 'type']);
+  }
+};
+
+/**
+ * Adds an authorized delegatee via a contract to perform
+ * the specified type of action
+ */
+actions.addAuthorization = async (payload) => {
+  const {
+    contract,
+    version,
+    action,
+    symbol,
+    type,
+    callingContractInfo,
+  } = payload;
+  if (api.assert(!callingContractInfo, 'this action cannot be invoked by a contract')) {
+    if (api.assert(contract
+      && contract.length > 3
+      && contract.length < 50
+      && version
+      && typeof version === 'number'
```

Comment:
the version of a contract is an integer so i'd recommend to make sure that the version passed by a contract/user is an integer (Number.isInteger(value))

Reply:
Ok cool